flang/lib/Lower/Bridge.cpp

```diff
@@ -395,6 +396,14 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     return cat == Fortran::common::TypeCategory::Derived;
   }
 
+  /// Insert a new block before \p block.  Leave the insertion point unchanged.
+  mlir::Block *insertBlock(mlir::Block *block) {
```

Comment:
mlir functions `splitBlock` and (one overload of) `createBlock` are complementary.  `createBlock` inserts a new block before a given block, and `splitBlock` inserts the new block after the given block.  Since "the given block" may actually be the first of multiple blocks, `createBlock` makes much more sense here.  Of course, where blocks are inserted doesn't affect actual code relationships between blocks.  But it is much easier to read and debug IR when blocks are placed in more natural locations.